ambry-api/src/main/java/com/github/ambry/router/Router.java

```diff
@@ -173,6 +186,20 @@ default CompletableFuture<String> putBlob(BlobProperties blobProperties, byte[]
     return future;
   }
 
+  /**
+   * Requests for a blob to be replicated asynchronously and returns a future that will eventually contain information
+   * about whether the request succeeded or not.
+   * @param blobId The ID of the blob to be replicated.
+   * @param serviceId The service ID of the service replicating the blob. This can be null if unknown.
+   * @param sourceDataNode The source {@link DataNodeId} to get the blob from.
+   * @return A future that would contain information about whether the replicateBlob succeeded or not, eventually.
+   */
+  default CompletableFuture<Void> replicateBlob(String blobId, String serviceId, DataNodeId sourceDataNode) {
+    CompletableFuture<Void> future = new CompletableFuture<>();
+    replicateBlob(blobId, serviceId, sourceDataNode, CallbackUtils.fromCompletableFuture(future), null);
+    return future;
```

Comment:
bit confused here. All 3 routers below - non-blocking, in-mem, perf - don't support replicateBlob. So which router actually supports it ?

Reply:
This is the PR to add the interface. The next PR will add the implementation in Non Blocking Router.

But Justin suggested don't add interface and don't open up it to the client. And we only use the ReplicateBlobRequest internally.